LLama/ChatSession.cs

```diff
@@ -159,15 +159,15 @@ public async IAsyncEnumerable<string> ChatAsync(string prompt, IInferenceParams?
                 InteractiveExecutorState state = (InteractiveExecutorState)executor.GetStateData();
                 prompt = state.IsPromptRun
                     ? HistoryTransform.HistoryToText(History)
```

Comment:
There's an unexpected behaviour here. If the user input the prompt for the first time run, what he/she want is treat it as a raw text instead of transformed text. You could run the example `ChatSessionWithRoleName` and set break point to check it. 

It's not your fault, because the abstraction here makes the boundary vague. Some logic of executors and chat session need to be refactored to completely fix it. However a quick fix for it now will be better. :)

Reply:
Yeah, I see what you're saying. Do you think the change to the overlay that uses the `ChatHistory` is ok to go in by itself? The following messages a user sends in need the template from the `IHistoryTransform` implementation applied so that the model has the appropriate tokens to continue the conversation.

Reply:
I think it's better to see if the history is empty before processing the prompt here. If the history is empty, it indicates that the current prompt is the first input, therefore we don't modify it. Otherwise we could resume from history.

Reply:
If a session was loaded wouldn't the history be duplicated in the state? I think this is a problem that's created because the actual text is not saved when a session is saved. I'm working on a proposal for the `ChatSession` class that will communicate the behavior of the class better through its interface, and will have some more guardrails in place so it can't be used in unexpected ways.

I'll send that in as a separate PR and close this. I don't think we'll get this right without breaking changes.

Reply:
Actually we may also refactor some parts recently. If you'd like to make some aggressive break changes, you could also open a PR to the preview branch, so that you wouldn't feel limited. :)